google/cloud/internal/curl_impl.cc

```diff
@@ -528,49 +532,40 @@ StatusOr<int> CurlImpl::PerformWork() {
       if (in_multi_) {
         // In the extremely unlikely case that removing the handle from CURLM*
         // was an error, return that as a status.
-        multi_remove_status = AsStatus(
-            curl_multi_remove_handle(multi_.get(), handle_.handle_.get()),
-            __func__);
+        multi_remove_result =
+            curl_multi_remove_handle(multi_.get(), handle_.handle_.get());
         in_multi_ = false;
       }
 
-      TRACE_STATE() << ", status=" << status << ", remaining=" << remaining
+      TRACE_STATE() << ", status="
+                    << CurlHandle::AsStatus(msg->data.result, __func__)
+                    << ", remaining=" << remaining
                     << ", running_handles=" << running_handles
                     << ", multi_remove_status=" << multi_remove_status << "\n";
 
       // Ignore errors when closing the handle. They are expected because
       // libcurl may have received a block of data, but the WriteCallback()
       // (see above) tells libcurl that it cannot receive more data.
       if (closing_) continue;
-      if (!status.ok()) return status;
-      if (!multi_remove_status.ok()) return multi_remove_status;
+      if (result != CURLM_OK)
+        return CurlHandle::AsStatus(msg->data.result, __func__);
+      if (!multi_remove_status.ok())
+        return AsStatus(multi_remove_result, __func__);
     }
   }
   TRACE_STATE() << ", running_handles=" << running_handles << "\n";
   return running_handles;
 }
 
 Status CurlImpl::WaitForHandles(int& repeats) {
-  int const timeout_ms = 1;
-  std::chrono::milliseconds const timeout(timeout_ms);
+  int const timeout_ms = 1000;
   int numfds = 0;
   CURLMcode result =
-      curl_multi_wait(multi_.get(), nullptr, 0, timeout_ms, &numfds);
+      curl_multi_poll(multi_.get(), nullptr, 0, timeout_ms, &numfds);
```

Comment:
`curl_multi_poll()` was added in CURL 7.66.0, and we currently say we support CURL >= 7.47.0 (in fact we test with 7.29.0, for RHEL:7 / CentOS:7)